### ISSUE TITLE
RuntimeError: one of the variables needed for gradient computation has been modified by an inplace operation

### ISSUE THREAD

**Issue by cherrerajobs** [nicolai256/Stable-textual-inversion_win#17]
I'm getting this error when running v1-finetune.yaml with Stable Diffusion 1.4 checkpoint
`Variable._execution_engine.run_backward(                                                                            RuntimeError: one of the variables needed for gradient computation has been modified by an inplace operation: [torch.cuda.FloatTensor [4, 1024, 77]], which is output 0 of AsStridedBackward0, is at version 2; expected version 1 instead. Hint: enable anomaly detection to find the operation that failed to compute its gradient, with torch.autograd.set_detect_anomaly(True).`

Here's the full stack:

> Traceback (most recent call last):                                                                                        File "main.py", line 796, in <module>                                                                                     trainer.fit(model, data)                                                                                              File "D:\miniconda3\envs\ldm\lib\site-packages\pytorch_lightning\trainer\trainer.py", line 740, in fit                    self._call_and_handle_interrupt(                                                                                      File "D:\miniconda3\envs\ldm\lib\site-packages\pytorch_lightning\trainer\trainer.py", line 685, in _call_and_handle_interrupt                                                                                                                     return trainer_fn(*args, **kwargs)                                                                                    File "D:\miniconda3\envs\ldm\lib\site-packages\pytorch_lightning\trainer\trainer.py", line 777, in _fit_impl              self._run(model, ckpt_path=ckpt_path)                                                                                 File "D:\miniconda3\envs\ldm\lib\site-packages\pytorch_lightning\trainer\trainer.py", line 1199, in _run                  self._dispatch()                                                                                                      File "D:\miniconda3\envs\ldm\lib\site-packages\pytorch_lightning\trainer\trainer.py", line 1279, in _dispatch             self.training_type_plugin.start_training(self)                                                                        File "D:\miniconda3\envs\ldm\lib\site-packages\pytorch_lightning\plugins\training_type\training_type_plugin.py", line 202, in start_training                                                                                                      self._results = trainer.run_stage()                                                                                   File "D:\miniconda3\envs\ldm\lib\site-packages\pytorch_lightning\trainer\trainer.py", line 1289, in run_stage             return self._run_train()                                                                                              File "D:\miniconda3\envs\ldm\lib\site-packages\pytorch_lightning\trainer\trainer.py", line 1319, in _run_train            self.fit_loop.run()                                                                                                   File "D:\miniconda3\envs\ldm\lib\site-packages\pytorch_lightning\loops\base.py", line 145, in run                         self.advance(*args, **kwargs)                                                                                         File "D:\miniconda3\envs\ldm\lib\site-packages\pytorch_lightning\loops\fit_loop.py", line 234, in advance                 self.epoch_loop.run(data_fetcher)                                                                                     File "D:\miniconda3\envs\ldm\lib\site-packages\pytorch_lightning\loops\base.py", line 145, in run                         self.advance(*args, **kwargs)                                                                                         File "D:\miniconda3\envs\ldm\lib\site-packages\pytorch_lightning\loops\epoch\training_epoch_loop.py", line 193, in advance                                                                                                                        batch_output = self.batch_loop.run(batch, batch_idx)                                                                  File "D:\miniconda3\envs\ldm\lib\site-packages\pytorch_lightning\loops\base.py", line 145, in run                         self.advance(*args, **kwargs)                                                                                         File "D:\miniconda3\envs\ldm\lib\site-packages\pytorch_lightning\loops\batch\training_batch_loop.py", line 88, in advance                                                                                                                         outputs = self.optimizer_loop.run(split_batch, optimizers, batch_idx)                                                 File "D:\miniconda3\envs\ldm\lib\site-packages\pytorch_lightning\loops\base.py", line 145, in run                         self.advance(*args, **kwargs)                                                                                         File "D:\miniconda3\envs\ldm\lib\site-packages\pytorch_lightning\loops\optimization\optimizer_loop.py", line 215, in advance                                                                                                                      result = self._run_optimization(                                                                                      File "D:\miniconda3\envs\ldm\lib\site-packages\pytorch_lightning\loops\optimization\optimizer_loop.py", line 266, in _run_optimization                                                                                                            self._optimizer_step(optimizer, opt_idx, batch_idx, closure)                                                          File "D:\miniconda3\envs\ldm\lib\site-packages\pytorch_lightning\loops\optimization\optimizer_loop.py", line 378, in _optimizer_step                                                                                                              lightning_module.optimizer_step(                                                                                      File "D:\miniconda3\envs\ldm\lib\site-packages\pytorch_lightning\core\lightning.py", line 1652, in optimizer_step         optimizer.step(closure=optimizer_closure)                                                                             File "D:\miniconda3\envs\ldm\lib\site-packages\pytorch_lightning\core\optimizer.py", line 164, in step                    trainer.accelerator.optimizer_step(self._optimizer, self._optimizer_idx, closure, **kwargs)                           File "D:\miniconda3\envs\ldm\lib\site-packages\pytorch_lightning\accelerators\accelerator.py", line 336, in optimizer_step                                                                                                                        self.precision_plugin.optimizer_step(model, optimizer, opt_idx, closure, **kwargs)                                    File "D:\miniconda3\envs\ldm\lib\site-packages\pytorch_lightning\plugins\precision\precision_plugin.py", line 163, in optimizer_step                                                                                                              optimizer.step(closure=closure, **kwargs)                                                                             File "D:\miniconda3\envs\ldm\lib\site-packages\torch\optim\optimizer.py", line 88, in wrapper                             return func(*args, **kwargs)                                                                                          File "D:\miniconda3\envs\ldm\lib\site-packages\torch\autograd\grad_mode.py", line 28, in decorate_context                 return func(*args, **kwargs)                                                                                          File "D:\miniconda3\envs\ldm\lib\site-packages\torch\optim\adamw.py", line 92, in step                                    loss = closure()                                                                                                      File "D:\miniconda3\envs\ldm\lib\site-packages\pytorch_lightning\plugins\precision\precision_plugin.py", line 148, in _wrap_closure                                                                                                               closure_result = closure()                                                                                            File "D:\miniconda3\envs\ldm\lib\site-packages\pytorch_lightning\loops\optimization\optimizer_loop.py", line 160, in __call__                                                                                                                     self._result = self.closure(*args, **kwargs)                                                                          File "D:\miniconda3\envs\ldm\lib\site-packages\pytorch_lightning\loops\optimization\optimizer_loop.py", line 155, in closure                                                                                                                      self._backward_fn(step_output.closure_loss)                                                                           File "D:\miniconda3\envs\ldm\lib\site-packages\pytorch_lightning\loops\optimization\optimizer_loop.py", line 327, in backward_fn                                                                                                                  self.trainer.accelerator.backward(loss, optimizer, opt_idx)                                                           File "D:\miniconda3\envs\ldm\lib\site-packages\pytorch_lightning\accelerators\accelerator.py", line 311, in backward      self.precision_plugin.backward(self.lightning_module, closure_loss, *args, **kwargs)                                  File "D:\miniconda3\envs\ldm\lib\site-packages\pytorch_lightning\plugins\precision\precision_plugin.py", line 91, in backward                                                                                                                     model.backward(closure_loss, optimizer, *args, **kwargs)                                                              File "D:\miniconda3\envs\ldm\lib\site-packages\pytorch_lightning\core\lightning.py", line 1434, in backward               loss.backward(*args, **kwargs)                                                                                        File "D:\miniconda3\envs\ldm\lib\site-packages\torch\_tensor.py", line 307, in backward                                   torch.autograd.backward(self, gradient, retain_graph, create_graph, inputs=inputs)                                    File "D:\miniconda3\envs\ldm\lib\site-packages\torch\autograd\__init__.py", line 154, in backward                         Variable._execution_engine.run_backward(                                                                            RuntimeError: one of the variables needed for gradient computation has been modified by an inplace operation: [torch.cuda.FloatTensor [4, 1024, 77]], which is output 0 of AsStridedBackward0, is at version 2; expected version 1 instead. Hint: enable anomaly detection to find the operation that failed to compute its gradient, with torch.autograd.set_detect_anomaly(True).

**Comment by cherrerajobs** [comment]
Wrong model checkpoint used, the fix is to use the right model `text2img-large/model.ckpt`

**Comment by cherrerajobs** [comment]
Attempting to run it with https://ommer-lab.com/files/latent-diffusion/nitro/txt2img-f8-large/model.ckpt just causes a different set of errors:

> RuntimeError: Error(s) in loading state_dict for LatentDiffusion:                                                               size mismatch for model.diffusion_model.input_blocks.1.1.transformer_blocks.0.attn2.to_k.weight: copying a param with shape torch.Size([320, 1280]) from checkpoint, the shape in current model is torch.Size([320, 768]).                      size mismatch for model.diffusion_model.input_blocks.1.1.transformer_blocks.0.attn2.to_v.weight: copying a param with shape torch.Size([320, 1280]) from checkpoint, the shape in current model is torch.Size([320, 768]).                      size mismatch for model.diffusion_model.input_blocks.2.1.transformer_blocks.0.attn2.to_k.weight: copying a param with shape torch.Size([320, 1280]) from checkpoint, the shape in current model is torch.Size([320, 768]).                      size mismatch for model.diffusion_model.input_blocks.2.1.transformer_blocks.0.attn2.to_v.weight: copying a param with shape torch.Size([320, 1280]) from checkpoint, the shape in current model is torch.Size([320, 768]).                      size mismatch for model.diffusion_model.input_blocks.4.1.transformer_blocks.0.attn2.to_k.weight: copying a param with shape torch.Size([640, 1280]) from checkpoint, the shape in current model is torch.Size([640, 768]).                      size mismatch for model.diffusion_model.input_blocks.4.1.transformer_blocks.0.attn2.to_v.weight: copying a param with shape torch.Size([640, 1280]) from checkpoint, the shape in current model is torch.Size([640, 768]).                      size mismatch for model.diffusion_model.input_blocks.5.1.transformer_blocks.0.attn2.to_k.weight: copying a param with shape torch.Size([640, 1280]) from checkpoint, the shape in current model is torch.Size([640, 768]).                      size mismatch for model.diffusion_model.input_blocks.5.1.transformer_blocks.0.attn2.to_v.weight: copying a param with shape torch.Size([640, 1280]) from checkpoint, the shape in current model is torch.Size([640, 768]).                      size mismatch for model.diffusion_model.input_blocks.7.1.transformer_blocks.0.attn2.to_k.weight: copying a param with shape torch.Size([1280, 1280]) from checkpoint, the shape in current model is torch.Size([1280, 768]).                    size mismatch for model.diffusion_model.input_blocks.7.1.transformer_blocks.0.attn2.to_v.weight: copying a param with shape torch.Size([1280, 1280]) from checkpoint, the shape in current model is torch.Size([1280, 768]).                    size mismatch for model.diffusion_model.input_blocks.8.1.transformer_blocks.0.attn2.to_k.weight: copying a param with shape torch.Size([1280, 1280]) from checkpoint, the shape in current model is torch.Size([1280, 768]).                    size mismatch for model.diffusion_model.input_blocks.8.1.transformer_blocks.0.attn2.to_v.weight: copying a param with shape torch.Size([1280, 1280]) from checkpoint, the shape in current model is torch.Size([1280, 768]).                    size mismatch for model.diffusion_model.middle_block.1.transformer_blocks.0.attn2.to_k.weight: copying a param with shape torch.Size([1280, 1280]) from checkpoint, the shape in current model is torch.Size([1280, 768]).                      size mismatch for model.diffusion_model.middle_block.1.transformer_blocks.0.attn2.to_v.weight: copying a param with shape torch.Size([1280, 1280]) from checkpoint, the shape in current model is torch.Size([1280, 768]).                      size mismatch for model.diffusion_model.output_blocks.3.1.transformer_blocks.0.attn2.to_k.weight: copying a param with shape torch.Size([1280, 1280]) from checkpoint, the shape in current model is torch.Size([1280, 768]).                   size mismatch for model.diffusion_model.output_blocks.3.1.transformer_blocks.0.attn2.to_v.weight: copying a param with shape torch.Size([1280, 1280]) from checkpoint, the shape in current model is torch.Size([1280, 768]).                   size mismatch for model.diffusion_model.output_blocks.4.1.transformer_blocks.0.attn2.to_k.weight: copying a param with shape torch.Size([1280, 1280]) from checkpoint, the shape in current model is torch.Size([1280, 768]).                   size mismatch for model.diffusion_model.output_blocks.4.1.transformer_blocks.0.attn2.to_v.weight: copying a param with shape torch.Size([1280, 1280]) from checkpoint, the shape in current model is torch.Size([1280, 768]).                   size mismatch for model.diffusion_model.output_blocks.5.1.transformer_blocks.0.attn2.to_k.weight: copying a param with shape torch.Size([1280, 1280]) from checkpoint, the shape in current model is torch.Size([1280, 768]).                   size mismatch for model.diffusion_model.output_blocks.5.1.transformer_blocks.0.attn2.to_v.weight: copying a param with shape torch.Size([1280, 1280]) from checkpoint, the shape in current model is torch.Size([1280, 768]).                   size mismatch for model.diffusion_model.output_blocks.6.1.transformer_blocks.0.attn2.to_k.weight: copying a param with shape torch.Size([640, 1280]) from checkpoint, the shape in current model is torch.Size([640, 768]).                     size mismatch for model.diffusion_model.output_blocks.6.1.transformer_blocks.0.attn2.to_v.weight: copying a param with shape torch.Size([640, 1280]) from checkpoint, the shape in current model is torch.Size([640, 768]).                     size mismatch for model.diffusion_model.output_blocks.7.1.transformer_blocks.0.attn2.to_k.weight: copying a param with shape torch.Size([640, 1280]) from checkpoint, the shape in current model is torch.Size([640, 768]).                     size mismatch for model.diffusion_model.output_blocks.7.1.transformer_blocks.0.attn2.to_v.weight: copying a param with shape torch.Size([640, 1280]) from checkpoint, the shape in current model is torch.Size([640, 768]).                     size mismatch for model.diffusion_model.output_blocks.8.1.transformer_blocks.0.attn2.to_k.weight: copying a param with shape torch.Size([640, 1280]) from checkpoint, the shape in current model is torch.Size([640, 768]).                     size mismatch for model.diffusion_model.output_blocks.8.1.transformer_blocks.0.attn2.to_v.weight: copying a param with shape torch.Size([640, 1280]) from checkpoint, the shape in current model is torch.Size([640, 768]).                     size mismatch for model.diffusion_model.output_blocks.9.1.transformer_blocks.0.attn2.to_k.weight: copying a param with shape torch.Size([320, 1280]) from checkpoint, the shape in current model is torch.Size([320, 768]).                     size mismatch for model.diffusion_model.output_blocks.9.1.transformer_blocks.0.attn2.to_v.weight: copying a param with shape torch.Size([320, 1280]) from checkpoint, the shape in current model is torch.Size([320, 768]).                     size mismatch for model.diffusion_model.output_blocks.10.1.transformer_blocks.0.attn2.to_k.weight: copying a param with shape torch.Size([320, 1280]) from checkpoint, the shape in current model is torch.Size([320, 768]).                    size mismatch for model.diffusion_model.output_blocks.10.1.transformer_blocks.0.attn2.to_v.weight: copying a param with shape torch.Size([320, 1280]) from checkpoint, the shape in current model is torch.Size([320, 768]).                    size mismatch for model.diffusion_model.output_blocks.11.1.transformer_blocks.0.attn2.to_k.weight: copying a param with shape torch.Size([320, 1280]) from checkpoint, the shape in current model is torch.Size([320, 768]).                    size mismatch for model.diffusion_model.output_blocks.11.1.transformer_blocks.0.attn2.to_v.weight: copying a param with shape torch.Size([320, 1280]) from checkpoint, the shape in current model is torch.Size([320, 768]).

**Comment by cherrerajobs** [comment]
Okay so I spent the better part of the day trying to get this to work and I got it working on an Nvidia 10GB 3080 Size 448x448 images. Required a completely fresh clone of the files and a simple drop-in of Stable Diffusion 1.4. Everything else was making sure the conda environment was sane, I had to make sure to delete tensorflow as I was getting Keras errors not related to this project or its files as well.

The error above indicates user error.
